src/OpenIddict.Core/OpenIddictProvider.cs

```diff
@@ -234,6 +240,55 @@ public class OpenIddictProvider<TUser, TApplication> : OpenIdConnectServerProvid
             context.HandleResponse();
         }
 
+        public override async Task ProfileEndpoint([NotNull] ProfileEndpointContext context) {
+            var manager = context.HttpContext.RequestServices.GetRequiredService<OpenIddictManager<TUser, TApplication>>();
+
+            var principal = context.AuthenticationTicket?.Principal;
```

Comment:
Actually, ASOS is supposed to guard against null tickets. So I guess I'm probably paranoid here :smile: